### DISCUSSION TITLE
Packaging (ex: as a standalone executable file)

### DISCUSSION THREAD

**Discussion by ndonkoHenri** [seleniumbase/SeleniumBase#1895] [Q&A] [answered]
Hello, I will please like to know how I could package my SeleniumBase program with something like Pyinstaller for example in order to make it an executable file.

This is to avoid always having to open my terminal and activate my Venvs. And mainly for portability.. running my script on any PC, just by double clicking the executable file.

Thanks in advance for any guide or help on this.

**Comment by mdmintz** [comment, marked as answer]
Standard [PyInstaller](https://github.com/pyinstaller/pyinstaller) instructions will work. Tested with a TKinter app that ran a SeleniumBase script.
(See https://github.com/seleniumbase/SeleniumBase/issues/1075#issuecomment-973021116 for the thread.)
For convenience, you may want to use one of the [SeleniumBase Syntax Formats](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md) that can be run with pure ``python``, instead of ``pytest``. That may make things easier.

**Reply by ndonkoHenri** [reply]
Ok @mdmintz, I am on this now.
The two main issues are, I don't want the person who receives the pyinstaller made EXE file to have to install python + sbase + pytest.
So, my question is, is it possible at this moment to run a sbase test just from code without having to use `os.system` or anything similar?
So that, when my user presses a button in my GUi app, I could just call this method (ex: `self.run_sb_test`) and that's all? Briefly, any other way which will give my user the possibility of NOT installing python + sbase + pytest.

**Reply by mdmintz** [reply]
That depends on what ``Pyinstaller`` allows and how it packages executables. I'm not familiar with the internals of ``Pyinstaller``, but I have seen it and used it to make executable files, but it may not abstract everything that people want abstracted.
